R/functions_netCDF.R

```diff
@@ -1,98 +1,278 @@
-#' @param file A character string. The file path of the \var{netCDF} file to be
+#'  The user provides an array and specifies spatial information, dimensionality,
+#'  and metadata in order to create an empty netcdf.
+#'
+#'  The netCDF always includes at least two dimensions: Latitude and Longitude.
+#'  Information about these dimensions are taken from the \var{locations},
+#'  \var{isGridded}, \var{crs}, and \var{grid} arguments. One of \var{locations}
+#'  or \var{grid} needs to be provided, but not both. If \var{locations} is a matrix
+#'  or data.frame, \var{crs} needs to be provided.
+#'  Additional dimensions are added by setting \var{has_T_timeAxis} and/or
+#'  \var{has_Z_verticalAxis} to \code{TRUE}. If both flags are set to
+#'  \code{FALSE} then the netCDF is two dimensions with one or more variables.
+#'  The number of variables is set in the \var{var_attributes} argument. If either
+#'  \var{has_T_timeAxis} \emph{or} \var{has_Z_verticalAxis} is set to \code{TRUE} then a
+#'  third dimension is added to the netCDF. If \emph{both} \var{has_T_timeAxis}
+#'  and \var{has_Z_verticalAxis} are set to TRUE, then the netCDF will have four
+#'  dimensions, with time as the third, and depth as the fourth. Information about
+#'  these dimensions is set in the respective \var{bounds} and \var{attributes} arguments.
+#'  \var{has_T_timeAxis} and/or \var{has_Z_verticalAxis} is set to \code{TRUE}
+#'  the netCDF can \emph{only} have one variable.
+#'  The \var{data} object is used to set up the size and dataType in the netCDF. The metadata
+#'  needs to match the data. This object contains the data that you intend to
+#'  populate the netCDF wtih. The array should be set up so that each row is a site's info,
+#'  and each column is a value for variables, time or depth. If the netcdf is 4d
+#'  (time and depth == \code{TRUE}, then need to have a 3-d data array,
+#'  where each additional dimension contains values for each depth.
+#'
+#'
+#' @param data A numeric array.
+#' @param has_T_timeAxis A logical value. Indicates that the netCDF created will
+#'     have a third, time dimension.
+#' @param has_Z_verticalAxis A logical value. Indicates that the netCDF created will
+#'     have a veritical (e.g. soil profile depths) dimension. If \var{has_T_timeAxis}
+#'     is set to \code{FALSE} then the Z axis will be the third dimension.
+#'     If \var{has_T_timeAxis} is set to \code{TRUE} then the Z axis will be the
+#'     fourth dimension.
+#' @param time_bounds A numeric string that continuously lists the lower and upper
+#'    bounds of each time dimension value. In the absence of a time dimension,
+#'    this argument can be used to define the calculation period of the variable(s).
+#' @param vert_bounds A numeric string that continuously lists the lower and upper
+#'    bounds of each vertical dimension value.
+#' @param var_attributes A list of named character strings defining the variables
```

Comment:
I think we should encourage using CMIP6 standard names where available and link to variable lists, e.g., the  [day](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_day.json) and [Eday](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_Eday.json) tables for time-varying variables and to [fx](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_fx.json) and [Efx](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_Efx.json) for time-invariant variables.

Reply:
fixed